collect_app/src/androidTest/java/org/odk/collect/android/feature/formentry/BackgroundAudioRecordingTest.java

```diff
@@ -0,0 +1,86 @@
+import org.odk.collect.android.support.pages.FormEndPage;
+import org.odk.collect.android.support.pages.FormEntryPage;
+import org.odk.collect.audiorecorder.recording.AudioRecorderViewModelFactory;
+import org.odk.collect.audiorecorder.testsupport.StubAudioRecorderViewModel;
+
+import java.io.File;
+import java.io.IOException;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.odk.collect.android.support.FileUtils.copyFileFromAssets;
+
+@RunWith(AndroidJUnit4.class)
+public class BackgroundAudioRecordingTest {
+
+    private StubAudioRecorderViewModel stubAudioRecorderViewModel;
+
+    public final TestDependencies testDependencies = new TestDependencies() {
+        @Override
+        public AudioRecorderViewModelFactory providesAudioRecorderViewModelFactory(Application application) {
+            return new AudioRecorderViewModelFactory(application) {
+                @Override
+                public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
+                    if (stubAudioRecorderViewModel == null) {
+                        try {
+                            File stubRecording = File.createTempFile("test", ".m4a");
+                            stubRecording.deleteOnExit();
+
+                            copyFileFromAssets("media/test.m4a", stubRecording.getAbsolutePath());
+                            stubAudioRecorderViewModel = new StubAudioRecorderViewModel(stubRecording.getAbsolutePath());
+                        } catch (IOException e) {
+                            throw new RuntimeException(e);
+                        }
+                    }
+
+                    return (T) stubAudioRecorderViewModel;
+                }
+            };
+        }
+    };
+
+    public final CollectTestRule rule = new CollectTestRule();
+
+    @Rule
+    public final RuleChain chain = TestRuleChain.chain(testDependencies)
+            .around(GrantPermissionRule.grant(Manifest.permission.RECORD_AUDIO))
+            .around(rule);
+
+    @Test
+    public void whenBackgroundAudioRecordingEnabled_fillingOutForm_recordsAudio() {
+        FormEntryPage formEntryPage = rule.mainMenu()
+                .enableBackgroundAudioRecording()
+                .copyForm("one-question.xml")
+                .startBlankForm("One Question");
+        assertThat(stubAudioRecorderViewModel.isRecording(), is(true));
+
+        FormEndPage formEndPage = formEntryPage
+                .inputText("123")
+                .swipeToEndScreen();
+        assertThat(stubAudioRecorderViewModel.isRecording(), is(true));
+
+        formEndPage.clickSaveAndExit();
+        assertThat(stubAudioRecorderViewModel.isRecording(), is(false));
+
+        assertThat(stubAudioRecorderViewModel.getLastRecording(), notNullValue());
+        assertThat(stubAudioRecorderViewModel.getLastRecording().exists(), is(true));
```

Comment:
Right now we're just checking the recording exists on the disk. Once the form spec has been agreed on we can change this.